ledger/eval/eval.go

```diff
@@ -193,6 +206,56 @@ func (x *roundCowBase) lookup(addr basics.Address) (ledgercore.AccountData, erro
 	return ad, err
 }
 
+// balanceRound reproduces the way that the agreement package finds the round to
+// consider for online accounts.
+func (x *roundCowBase) balanceRound() (basics.Round, error) {
+	phdr, err := x.BlockHdr(agreement.ParamsRound(x.rnd))
+	if err != nil {
+		return 0, err
+	}
+	agreementParams := config.Consensus[phdr.CurrentProtocol]
+	return agreement.BalanceRound(x.rnd, agreementParams), nil
+}
+
+// lookupAgreement returns the online accountdata for the provided account address. It uses an internal cache
+// to avoid repeated lookups against the ledger.
+func (x *roundCowBase) lookupAgreement(addr basics.Address) (basics.OnlineAccountData, error) {
+	if accountData, found := x.onlineAccounts[addr]; found {
+		return accountData, nil
+	}
+
+	brnd, err := x.balanceRound()
+	if err != nil {
+		return basics.OnlineAccountData{}, err
+	}
+	ad, err := x.l.LookupAgreement(brnd, addr)
+	if err != nil {
+		return basics.OnlineAccountData{}, err
+	}
+
+	x.onlineAccounts[addr] = ad
+	return ad, err
+}
+
+// lookupAgreement returns the online accountdata for the provided account address. It uses an internal cache
```

Comment:
NIT: this comment is c/p from `lookupAgreement`

Reply:
Thanks, some sort of cut n paste error, I think.